addons/sale_project/models/sale_order_line.py

```diff
@@ -108,7 +108,7 @@ def create(self, vals_list):
             if line.state == 'sale' and not line.is_expense:
                 line.sudo()._timesheet_service_generation()
                 # if the SO line created a task, post a message on the order
-                if line.task_id:
+                if line.task_id and line.project_id:
```

Comment:
Is this change fixing anything?
It prevents the message from being posted when the template is configured to create a task in an existing project (line has no project_id), and the message will still get posted multiple times when the template is configured to create a project and a task.
Am I missing something?

Reply:
@MissingNoShiny , Right, done the changes. the issue only occurs when a product is added to the fsm task so fixed it accordingly. Thanks